code/parachain/frame/cosmwasm/src/mapping.rs

```diff
@@ -0,0 +1,92 @@
+//! this simply maps `ibc-rs` and `cosmwasm-std` types back and forth
+
+use cosmwasm_vm::cosmwasm_std::{IbcChannel, IbcChannelOpenMsg, IbcEndpoint, IbcOrder as CwOrder};
+use ibc::core::{
+	ics04_channel::{
+		channel::{Counterparty, Order as IbcOrder},
+		error::Error as IbcError,
+		Version as IbcVersion,
+	},
+	ics24_host::identifier::{ChannelId, ConnectionId, PortId},
+};
+
+use crate::Config;
+pub fn ibc_to_cw_channel_open<T: Config + Send + Sync>(
+	channel_id: &ChannelId,
+	port_id: &PortId,
+	counterparty: &Counterparty,
+	order: IbcOrder,
+	version: &IbcVersion,
+	connection_hops: &[ConnectionId],
+) -> Result<IbcChannelOpenMsg, IbcError> {
+	Ok(IbcChannelOpenMsg::OpenInit {
+		channel: IbcChannel::new(
+			IbcEndpoint { channel_id: channel_id.to_string(), port_id: port_id.to_string() },
+			IbcEndpoint {
+				port_id: counterparty.port_id.to_string(),
+				channel_id: counterparty.channel_id.expect("channel").to_string(),
+			},
+			map_order(order)?,
+			version.to_string(),
+			connection_hops
+				.get(0)
+				.expect("by spec there is at least one connection; qed")
+				.to_string(),
+		),
+	})
+}
+
+pub fn map_order(order: IbcOrder) -> Result<CwOrder, IbcError> {
```

Comment:
This fn always returns Ok, is that intended?

Reply:
will remove. before that Centauri had 3 values and Coswasm has 2. Now both has 2.

Reply:
fixing